### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by jeromekelleher** [sgkit-dev/bio2zarr#174]
In principle this should work fine on Windows. I made a brief effort to get things working via GitHub actions, but hit some immediate issues with file paths in the Git checkout.

I'm very happy to support Windows, but would need some help from someone motivated to get things working initially.

**Comment by pmoris** [comment]
Seeing as we just added bio2zarr as a dependency for our dapcy package after sgkit[vcf] support was deprecated (see https://github.com/sgkit-dev/sgkit/pull/1264#issuecomment-2842334391), 

I quickly tried installing the package using `pip` (inside a miniforge environment) on my Windows system and ran into:

```
      RuntimeError: Required library 'hts-3.dll' not found. Please install htslib first.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cyvcf2
Successfully built asciitree
Failed to build cyvcf2
ERROR: Failed to build installable wheels for some pyproject.toml based projects (cyvcf2)
```

Looking at https://github.com/brentp/cyvcf2 and some of the [issues](https://github.com/brentp/cyvcf2/issues?q=windows), windows installations do not appear to be trivial.

Anything else I can try out to get this underway? I can try opening an issue on cyvcf2 as well perhaps.

( We're just submitting our tool to a journal and didn't realise until now that moving to bio2zarr would mean dropping windows support though, so we're a bit worried here. ) 

@alejocrojo09

**Comment by jeromekelleher** [comment]
Sorry about this @pmoris  - I would would love to provide an easy to install and straightforward way of supporting VCF through bio2zarr, but unfortunately this is not trivial because of our dependencies. Cyvcf2 isn't supported on Windows, and I think it's going to be hard to get that in place. We definitely need to use htslib for VCF parsing, so that's not going to change any time soon.

My only suggestion for Windows users is to switch to WSL for VCF conversion, which works pretty well.

Another approach for you may be to just require VCF Zarr as input rather than VCF, and ask your users to do the conversion (pointing to vcf2zarr docs)? That would make things simpler to explain, and wouldn't require you to have a direct dependency on bio2zarr (and therefore let you support Windows properly).

We also used cyvcf2 in sgkit, so the same problems would have been present on Windows there.

**Comment by pmoris** [comment]
Fair enough! We'll just provide the WSL and pre-created Zarr as suggestions for our Windows users then, thanks for the suggestions!

> We also used cyvcf2 in sgkit, so the same problems would have been present on Windows there.

I figured as much; that's on us for not checking other platforms sooner ;)

> Cyvcf2 isn't supported on Windows, and I think it's going to be hard to get that in place. 

I suspect the same... regardless, if I can be of any help for this in the future, I'll do my best!

**Comment by jeromekelleher** [comment]
The basic package is now supported on Windows, and everything seems to work OK except VCF support. I'm going to close this as done, noting that until cyvcf2 works on Windows, we can't do anything about supporting VCF.
